src/Task.ts

```diff
@@ -237,6 +242,16 @@ export class Task {
             description = description.replace(this.blockLinkRegex, '').trim();
         }
 
+        // check for any number of subtags on the global filter
+        let subTags: string = '';
+        const tagsMatch = description.match(Task.subTagRegex);
+        if (tagsMatch !== null) {
+            if (tagsMatch[1] === globalFilter) {
+                subTags =
+                    typeof tagsMatch[2] !== 'undefined' ? tagsMatch[2] : ''; // keep just the subtag(s)
+            }
+        }
+
```

Comment:
All the changes to Task need tests please.

The fact that the existing tests pass confirms that the changes have not broken existing behaviour, which is good, but they don't show the intention of this new code, and don't protect this new code from being broken by future changes to the code.

There is some info on running and maintaining the tests here:
https://github.com/obsidian-tasks-group/obsidian-tasks/blob/main/CONTRIBUTING.md#maintaining-the-tests

Please do ask if you would like to chat about this, if it's all new to you.